controllers/temprole_controller.py

```diff
@@ -123,6 +124,32 @@ async def view_temproles(user: User, interaction: Interaction):
 
         await interaction.response.send_message(response, ephemeral=True)
 
+    @staticmethod
+    async def view_users(role: Role, interaction: Interaction):
+        temprole_users = DB().get_temprole_users(role.id, interaction.guild_id)
```

Comment:
This one I'm going to call a "nit" because I don't think it will come up for our use case for a while, but it's something to keep in mind for the future.

In general with pagination, you don't want to retrieve all the results at once. Instead you want to retrieve a single page of results at a time with the correct offset. The reason for this is that as the number of results grows and grows, that's more information that you'd have to keep in memory at a given time. 

With your current implementation, if a mod asks to list all the Good Morning T2s then the bot will keep 500 temples in memory until that moderator hits "dismiss message" (or the timeout that discordpy imposes for views). If multiple mods ask to view temproles at the same time, this scales based on the number of temproles being listed rather than a constant 10 * number of mods listing in parallel.

Again, this is unlikely to cause a problem right now (500 isn't _that_ many to a computer nowadays) but wanted to point this out as a potential improvement if you're feeling ambitious. The way I'd likely refactor this is to provide the `PaginationEmbed` with a `get_next_page(current_page_number)` callback that is passed in at point of instantiation so that the view remains generic as you have it now.